File: src/main/java/de/notjansel/datapacksync/commands/copy.java
package de.notjansel.datapacksync.commands;

import de.notjansel.datapacksync.Datapacksync;
import de.notjansel.datapacksync.threading.CopyThread;
import io.papermc.paper.datapack.Datapack;
import org.bukkit.ChatColor;
import org.bukkit.World;
import org.bukkit.command.*;
import org.checkerframework.checker.units.qual.C;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.*;
import java.nio.file.*;
import java.util.ArrayList;
import java.util.List;

import static de.notjansel.datapacksync.Datapacksync.*;

public class copy implements CommandExecutor, TabCompleter {

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        CopyThread thread = new CopyThread(sender, args);
        thread.start();
        return true;
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String alias, @NotNull String[] args) {
        List<String> tabComplete = new ArrayList<>();
        if (args.length == 1) {
            File directoryPath = new File(Datapacksync.serverpath + "/downloads/");
            for (String filename : directoryPath.list()
                 ) {
                tabComplete.add(filename);
            }
        }


        return tabComplete;
    }
}

File: src/main/java/de/notjansel/datapacksync/commands/Copy.kt
package de.notjansel.datapacksync.commands

import de.notjansel.datapacksync.Datapacksync
import de.notjansel.datapacksync.threading.CopyThread
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import java.io.File

class Copy : CommandExecutor, TabCompleter {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        val thread = CopyThread(sender, args)
        thread.start()
        return true
    }

    override fun onTabComplete(sender: CommandSender, command: Command, alias: String, args: Array<String>): List<String>? {
        val tabComplete: MutableList<String> = ArrayList()
        if (args.size == 1) {
            val directoryPath = File(Datapacksync.serverpath + "/downloads/")
            for (filename in directoryPath.list()) {
                tabComplete.add(filename)
            }
        }
        return tabComplete
    }
}